pytest_pyvista.py

```diff
@@ -1,21 +1,168 @@
 # -*- coding: utf-8 -*-
 
+import os
+import pathlib
+import platform
+import warnings
+
 import pytest
+import pyvista
+from pyvista._vtk import VTK9
 
 
 def pytest_addoption(parser):
-    group = parser.getgroup('pyvista')
+    """Adds new flag options to the pyvista plugin."""
+
+    group = parser.getgroup("pyvista")
+    group.addoption(
+        "--reset_image_cache",
+        action="store_true",
+        help="Reset the images in the PyVista cache.",
+    )
     group.addoption(
-        '--foo',
-        action='store',
-        dest='dest_foo',
-        default='2022',
-        help='Set the value for the fixture "bar".'
+        "--ignore_image_cache", action="store_true", help="Ignores the image cache."
     )
+    group.addoption(
+        "--fail_extra_image_cache",
+        action="store_true",
+        help="Enables failure if image cache does not exist.",
+    )
+    parser.addini("HELLO", "Dummy pytest.ini setting")
+
+
+class VerifyImageCache:
+    """Control image caching for testing.
+
+    Image cache files are names according to ``test_name``.
+    Multiple calls to an instance of this class will append
+    `_X` to the name after the first one.  That is, files
+    ``{test_name}``, ``{test_name}_1``, and ``{test_name}_2``
+    will be saved if called 3 times.
+
+    Parameters
+    ----------
+    test_name : str
+        Name of test to save.  Sets name of image cache file.
+
+    """
+
+    reset_image_cache = False
+    ignore_image_cache = False
+    fail_extra_image_cache = False
+
+    high_variance_tests = False
+    windows_skip_image_cache = False
+    macos_skip_image_cache = False
+
+    def __init__(
+        self,
+        test_name,
+        *,
+        cache_dir=None,
+        error_value=500,
+        warning_value=200,
+        var_error_value=1000,
+        var_warning_value=1000,
+    ):
+        self.test_name = test_name
+
+        if cache_dir is None:
+            # Reset image cache with new images
+            this_path = pathlib.Path(__file__).parent.absolute()
+            self.cache_dir = os.path.join(this_path, "image_cache")
+        else:
+            self.cache_dir = cache_dir
+
+        if not os.path.isdir(self.cache_dir):
+            os.mkdir(self.cache_dir)
 
-    parser.addini('HELLO', 'Dummy pytest.ini setting')
+        self.error_value = error_value
+        self.warning_value = warning_value
+        # self.high_variance_tests = high_variance_tests
+        # self.windows_skip_image_cache = windows_skip_image_cache
+        # self.macos_skip_image_cache = macos_skip_image_cache
+        self.var_error_value = var_error_value
+        self.var_warning_value = var_warning_value
 
+        self.skip = False
```

Comment:
Do you want to explicitly add the parameter `skip` in the `__init__()` signature?

Reply:
The usage of skip is usually after instantiation since it is instantiated in a fixture in this package.  I assume the user will use it from the fixture typically.  We could add a getter/setter for skip so that it is better documented.

Reply:
@MatthewFlamm Agreed, that would be even better! Yes, this can be done in an other PR as well as the other properties.

Reply:
I agree, all attributes of the class should be accessed through getters and setters. I also agree that this should be done in another PR, to keep the size of this one reasonable. I opened an issue regarding this: #4.